klayout/drc/rule_decks/sab.drc

```diff
@@ -0,0 +1,180 @@
+  poly_sab.forget
+
+  # Rule SB.13: Min. area (um2) is 2µm².
+  logger.info('Executing rule SB.13')
+  sb13_l1 = sab_out_otp.with_area(nil, 2.um)
+  sb13_l1.output('SB.13', 'SB.13 : Min. area (um2): 2µm²')
+  sb13_l1.forget
+
+  # Rule SB.14a: Space from unsalicided Nplus Poly2 to unsalicided Pplus Poly2.
+  ## (Unsalicided Nplus Poly2 must not fall within a square of 0.56um x 0.56um
+  ## at unsalicided Pplus Poly2 corners) is 0.56µm.
+  logger.info('Executing rule SB.14a')
+  sab_poly_nplus = sab_poly.and(nplus)
+  sab_poly_pplus = sab_poly.and(pplus)
+  sb14a_butted_sab_poly = sab_poly_pplus.edges.inside(sab_poly_nplus.edges)
+  sb14a_l1 = sab_poly_nplus.separation(sab_poly_pplus, 0.56.um, square).polygons(0.001.um)
+                           .not_interacting(sb14a_butted_sab_poly)
+  sb14a_l1.output('SB.14a', "SB.14a : Space from unsalicided Nplus Poly2 to unsalicided Pplus Poly2.
+                  (Unsalicided Nplus Poly2 must not fall within a square of 0.56um x 0.56um
+                  at unsalicided Pplus Poly2 corners). : 0.56µm")
+  sb14a_l1.forget
+  sb14a_butted_sab_poly.forget
+  sab_poly_pplus.forget
+
+  # Rule SB.14b: Space from unsalicided Nplus Poly2 to P-channel gate.
+  ## (Unsalicided Nplus Poly2 must not fall within a square of 0.56um x 0.56um
+  ## at P-channel gate corners) is 0.56µm.
+  logger.info('Executing rule SB.14b')
+  sb14b_l1 = sab_poly_nplus.separation(pgate, 0.56.um, square)
+  sb14b_l1.output('SB.14b', "SB.14b : Space from unsalicided Nplus Poly2 to P-channel gate.
+                    (Unsalicided Nplus Poly2 must not fall within a square of 0.56um x 0.56um
+                    at P-channel gate corners): 0.56µm")
+  sb14b_l1.forget
+  sab_poly_nplus.forget
+
+  # Rule SB.15a: Space from unsalicided Poly2 to unrelated Nplus/Pplus. is 0.18µm
+  logger.info('Executing rule SB.15a')
+  sb15a_np = nplus.or(pplus).not_overlapping(sab_poly)
+  sb15a_l1 = sab_poly.separation(sb15a_np, 0.18.um, euclidian)
+  sb15a_l1.output('SB.15a', 'SB.15a : Space from unsalicided Poly2 to unrelated Nplus/Pplus: 0.18µm')
+  sb15a_l1.forget
+  sb15a_np.forget
+  sab_poly.forget
+
+  # Rule SB.15b: Space from unsalicided Poly2 to unrelated Nplus/Pplus along Poly2 line is 0.32 um.
+  logger.info('Executing rule SB.15b')
+  sb15b_np = nplus.or(pplus).interacting(poly2.interacting(sab_out_otp))
+  sb15b_poly = poly2.overlapping(sb15b_np).and(sab_out_otp)
+  sb15b_l1 = sb15b_poly.separation(sb15b_np, 0.32.um, euclidian)
+  sb15b_l1.output('SB.15b', 'SB.15b : Space from unsalicided Poly2 to unrelated Nplus/Pplus: 0.32µm')
+  sb15b_l1.forget
+  sb15b_np.forget
+  sb15b_poly.forget
+
+  # Rule SB.16: SAB layer cannot exist on 3.3V and 5V/6V CMOS transistors'
+  ## Poly and COMP area of the core circuit (Excluding the transistors used for ESD purpose).
+  ## It can only exist on CMOS transistors marked by LVS_IO, OTP_MK, ESD_MK layers.
+  logger.info('Executing rule SB.16')
+  sb16_exclude = sab_out_otp.outside(lvs_io.or(esd_mk))
```

Comment:
looks like rule `SB.12` is also computing `sab_out_otp.outside(esd_mk)` should we reuse it here?